src/features/student/declaredPrograms/stores/declaredPrograms.store.test.ts

```diff
@@ -0,0 +1,64 @@
+import { useDeclaredProgramsStore } from '@/features/student/declaredPrograms/stores/declaredPrograms.store'
+import { BddTest } from '@avenirs-esr/avenirs-dsav/test-utils'
+import { createPinia, setActivePinia } from 'pinia'
+import { beforeEach, expect } from 'vitest'
+
+BddTest().given('a declared programs store', () => {
+  beforeEach(() => {
+    setActivePinia(createPinia())
+  })
+
+  BddTest().when('the store is initialized', () => {
+    BddTest().then('it should provide drawer state and methods', () => {
+      const store = useDeclaredProgramsStore()
+
+      expect(store.showAddDeclaredProgramDrawer).toBeDefined()
+      expect(store.displayAddDeclaredProgramDrawer).toBeDefined()
+      expect(store.hideAddDeclaredProgramDrawer).toBeDefined()
+    })
+
+    BddTest().then('it should have drawer initially hidden', () => {
+      const store = useDeclaredProgramsStore()
+
+      expect(store.showAddDeclaredProgramDrawer).toBe(false)
+    })
+  })
+
+  BddTest().when('displaying the add declared program drawer', () => {
+    BddTest().then('it should set showAddDeclaredProgramDrawer to true', () => {
+      const store = useDeclaredProgramsStore()
+
+      store.displayAddDeclaredProgramDrawer()
+
+      expect(store.showAddDeclaredProgramDrawer).toBe(true)
+    })
+  })
+
+  BddTest().when('hiding the add declared program drawer', () => {
+    BddTest().then('it should set showAddDeclaredProgramDrawer to false', () => {
+      const store = useDeclaredProgramsStore()
+      store.displayAddDeclaredProgramDrawer()
+
+      store.hideAddDeclaredProgramDrawer()
+
+      expect(store.showAddDeclaredProgramDrawer).toBe(false)
+    })
+  })
```

Comment:
```suggestion

    BddTest().and('displaying the add declared program drawer', () => {
      beforeEach(() => {
        store.displayAddDeclaredProgramDrawer()
      })
    
      BddTest().then('it should set showAddDeclaredProgramDrawer to true', () => {
        expect(store.showAddDeclaredProgramDrawer).toBe(true)
      })
    })

    BddTest().and('hiding the add declared program drawer', () => {
      beforeEach(() => {
        store.displayAddDeclaredProgramDrawer()
        store.hideAddDeclaredProgramDrawer()
      })
    
      BddTest().then('it should set showAddDeclaredProgramDrawer to false', () => {
        expect(store.showAddDeclaredProgramDrawer).toBe(false)
      })
    })
```

Se baser sur ça pour l'ensemble du fichier de test

Reply:
@p-beger pourquoi faire un `and` dans le `when` de l'initialization du store ?

Reply:
ça sert à rien le beforeEach si ton `and` ou `when` contient un seul then

```ts
BddTest().and('displaying the add declared program drawer', () => {
      beforeEach(() => {
        store.displayAddDeclaredProgramDrawer()
      })
    
      BddTest().then('it should set showAddDeclaredProgramDrawer to true', () => {
        expect(store.showAddDeclaredProgramDrawer).toBe(true)
      })
    })
```